libs/constrained-generators/src/Constrained/Base.hs

```diff
@@ -5745,21 +5751,53 @@ maxSpec (TypeSpec (NumSpecInterval _ hi) bad) = TypeSpec (NumSpecInterval Nothin
 -- Sized
 -- ================
 
-class Sized t where
+class Sized fn t where
   sizeOf :: t -> Integer
+  default sizeOf :: (HasSimpleRep t, Sized fn (SimpleRep t)) => t -> Integer
+  sizeOf = sizeOf @fn . toSimpleRep
+
   liftSizeSpec :: HasSpec fn t => SizeSpec fn -> [Integer] -> Specification fn t
+  default liftSizeSpec ::
+    ( HasSpec fn t
+    , HasSimpleRep t
+    , Sized fn (SimpleRep t)
+    , HasSpec fn (SimpleRep t)
+    , TypeSpec fn t ~ TypeSpec fn (SimpleRep t)
+    ) =>
+    SizeSpec fn -> [Integer] -> Specification fn t
+  liftSizeSpec sz cant = fromSimpleRepSpec $ liftSizeSpec sz cant
```

Comment:
Is there a reason for all these default methods? I I recall correctly, there are only 3 Sized instances. It looks like the default methods are trying to work when the SimpleRep of a type is Sized, so perhaps sequences could get there Sized instances from List?  is that the case here?

Reply:
This is to make sure it's easy to add new instances when you have a type with a `SimpleRep` that has `Sized`. The `StrictSeq` instance is the key example of this - deriving from the instance for `[a]`.